lib/auto-build.ts

```diff
@@ -43,19 +43,10 @@ export class AutoBuild extends Construct {
     const project = new codebuild.Project(this, 'Project', {
       source: props.repo.createBuildSource(this, true),
       environment: createBuildEnvironment(props.environment),
-      badge: true,
+      badge: props.repo.allowsBadge,
       buildSpec: props.buildSpec
     });
 
-    // not support in this version of the cdk
-    const cfnProject = project.node.tryFindChild('Resource') as Resource;
```

Comment:
Is this a regression?

Reply:
No. Basically, this is now supported in the CDK directly, so no need to use an L1 for it.

What is more, webhooks are not allowed to be used with CodeCommit as the source of a CodeBuild project, so we shouldn't be adding this unconditionally - instead, we should rely on `source: props.repo.createBuildSource(this, true),` to make the decision whether the webhook can be created or not.